### PR TITLE
Added code to validate file extension and updated S3 policy

### LINKED ISSUE #187: limit file type for s3 uploads
## Type of Issue:

- [x] Enhancement: New feature or request

#### Issue Creator Checklist

- [x] This issue has been thoroughly documented below; a developer should be able to understand the issue by reading it.

## Background

Currently the s3 bucket allows any file type to be uploaded. We have ClamAV scanning to scan and flag malicious files however during Pen Testing one of the actions taken is to upload an html file with Javascript code embedded in the html.  This is marked as a high finding on the Pen Test and must be resolved before promoting to production. This issue is to address that finding by limiting the file types that can be uploaded from within the application. 

## Proposed Approach

The UI should do a file type check based on the file extension. If the file selected to upload is not an allowed file type then the UI will flag the file and not allow a submit action. In addition to the react file type check the s3 bucket policy should be updated to only allow the same list of file types.

Another consideration brought up was on file scanning to add a delete action if the file is marked malicious rather than tagging it so it cant be downloaded.. this is probably a good idea. 

the onemac team implemented this in the macstack repo see https://github.com/CMSgov/macstack-spa-submission-form/pull/227 for how they implemented this.  We probably should just use the list they compiled for acceptable file types as our source of truth as well.

## Value

Satisfies pen test findings as well as provides a more secure bucket only allowing intended file types.

## AC:
- The application will limit file type uploads to the s3 bucket.  For this story, the goal is to disallow *.html uploads
- The s3 bucket policy will be updated to reflect the file type allowed list (is this possible?)

### PR DESCRIPTION
**Pull request by ayoneeyee**
## Purpose

Currently the s3 bucket allows any file type to be uploaded. We have ClamAV scanning to scan and flag malicious files however during Pen Testing one of the actions taken is to upload an html file with Javascript code embedded in the html. This is marked as a high finding on the Pen Test and must be resolved before promoting to production. This issue is to address that finding by limiting the file types that can be uploaded from within the application.

#### Linked Issues to Close

Closes #187 

## Approach

The UI should do a file type check based on the file extension. If the file selected to upload is not an allowed file type then the UI will flag the file and not allow a submit action. In addition to the react file type check the s3 bucket policy should be updated to only allow the same list of file types.

## Learning

Updated ReactJS code to limit unapproved file type extensions
Updated S3 bukcet policy to limit upload to approved files extensions only
Reviewed the following pages:
https://github.com/CMSgov/macstack-spa-submission-form/pull/227/files
https://github.com/CMSgov/macpro-quickstart-serverless/blob/master/services/ui-src/src/containers/NewAmendment.js

## Assorted Notes/Considerations
Here is the Cloudfront URL:
Application endpoint:  https://d2l4lyrxqnnn4.cloudfront.net

In order to reproduce issue. Do the following:
1) Go to the Cloudfront URLabove
2) Upload any jpeg or png file (JPEG and PNG files are allowed files)
3) Confirm file was successfully uploaded
4) To confirm a reject file, upload a .HTML file
5) confirm if file was uploaded or denied

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/uploads/serverless.yml [hunk 1]
```diff
@@ -119,6 +119,11 @@ resources:
                   s3:ExistingObjectTag/virusScanStatus:
                     - "CLEAN"
                   aws:PrincipalArn: !GetAtt BucketAVScanRole.Arn
+            - Action:
+                - "s3:PutObject"
+              Effect: "Deny"
+              Resource: !Sub ${AttachmentsBucket.Arn}/*.html
+              Principal: "*"
```
Comment (mdial89f) [inline services/uploads/serverless.yml:126]: You're using a whitelist in the react code, but a black list in the bucket policy.  I recommend we keep it consistent; if a whitelist is what we've been given, let's use it in both places.

See docs on [how to restrict buckets to certain file types](https://aws.amazon.com/premiumsupport/knowledge-center/s3-allow-certain-file-types/)
That link above has an example of applying an extension whitelist to an S3 bucket's policy.  Pasting below....:
```
{
      "Sid": "Stmt1464968483619",
      "Effect": "Deny",
      "Principal": "*",
      "Action": "s3:PutObject",
      "NotResource": [
        "arn:aws:s3:::DOC-EXAMPLE-BUCKET/*.jpg",
        "arn:aws:s3:::DOC-EXAMPLE-BUCKET/*.png",
        "arn:aws:s3:::DOC-EXAMPLE-BUCKET/*.gif"
      ]
    }
```

**Review by mdial89f — CHANGES_REQUESTED** [review]
See comment for requested changes.  
In addition, run prettier when able; style_checks is failing

File: services/ui-src/src/containers/NewAmendment.js [hunk 2]
```diff
@@ -45,6 +45,15 @@ export default function NewAmendment({ fileUpload }) {
 
     if (!validateFileAttachment(file)) return;
 
+    var validExt = ".png, .gif, .jpeg, .jpg";
```
Comment (dwhitecl) [inline services/ui-src/src/containers/NewAmendment.js:48]: @ayoneeyee as mentioned in the issue description. we should use the allowed file types as defined in the macstack pull request which resolved this issue within their repo
https://github.com/CMSgov/macstack-spa-submission-form/pull/227/files

File: services/uploads/serverless.yml [hunk 3]
```diff
@@ -119,6 +119,54 @@ resources:
                   s3:ExistingObjectTag/virusScanStatus:
                     - "CLEAN"
                   aws:PrincipalArn: !GetAtt BucketAVScanRole.Arn
+            - Action: "s3:PutObject"
+              Effect: Deny
+              Principal: "*"
+              Resource:
```
Comment (dwhitecl) [inline services/uploads/serverless.yml:125]: did you test this? based on this PR 
https://github.com/CMSgov/macstack-spa-submission-form/pull/227/files
You should only need the NotResource block starting on line 146

**Pull request closed without merging**